pilot/pkg/bootstrap/server.go

```diff
@@ -1136,12 +1133,10 @@ func (s *Server) grpcServerOptions() []grpc.ServerOption {
 		prometheus.UnaryServerInterceptor,
 	}
 
-	// EnableHandlingTimeHistogram() has a data race within it (reading/writing to its
-	// serverHandledHistogramEnabled). Using a package mutex to avoid data races.
-	// Should be re-examined with an updated revision of the go-grpc-prometheus.
-	globalMutex.Lock()
-	prometheus.EnableHandlingTimeHistogram()
-	globalMutex.Unlock()
+	// Running the prometheus global func once to avoid data races within it
+	runOnce.Do(func() {
```

Comment:
The placement of the method call itself seems wrong. We're enabling a global setting within what looks like a pure function that generates option values. 

Perhaps placing it at a higher call point will help with the race condition itself as well?

Reply:
Good idea